wallet-client/WalletClient.hs

```diff
@@ -0,0 +1,223 @@
+muxLocalInitiatorNetworkApplication Proxy tracer pInfoConfig =
+    simpleSingletonVersions
+      NodeToClientV_1
+      (NodeToClientVersionData { networkMagic = 0 })
+      (DictVersion nodeToClientCodecCBORTerm)
+
+  $ MuxInitiatorApplication $ \ptcl -> case ptcl of
+      LocalTxSubmissionPtcl -> \channel -> do
+        txv <- newEmptyTMVarM @_ @(GenTx blk)
+        runPeer
+          nullTracer
+          localTxSubmissionCodec
+          channel
+          (localTxSubmissionClientPeer
+              (txSubmissionClient
+                @(GenTx blk)
+                (contramap condense tracer) txv))
+
+      ChainSyncWithBlocksPtcl -> \channel ->
+        runPeer
+          nullTracer
+          (localChainSyncCodec @blk pInfoConfig)
+          channel
+          (chainSyncClientPeer
+            $ chainSyncClient
+              (contramap condense tracer))
+
+
+data TraceTx tx reject
+  = TraceTxAccept tx
+  | TraceTxReject tx reject
+  deriving Show
+
+instance (Condense tx, Condense reject) => Condense (TraceTx tx reject) where 
+    condense (TraceTxAccept tx)        = "TxAccept "   <> condense tx
+    condense (TraceTxReject tx reject) = "TxRejected " <> condense tx
+                                                       <> " "
+                                                       <> condense reject
+
+
+-- | A 'LocalTxSubmissionClient' that submits transactions reading them from
+-- a 'TMVar'.  A real implementation should use a better synchronisation
+-- primitive.  This demo creates and empty 'TMVar' in
+-- 'muxLocalInitiatorNetworkApplication' above and never fills it with a tx.
+--
+txSubmissionClient
+  :: forall tx reject m.
+     ( Monad    m
+     , MonadSTM m
+     )
+  => Tracer m (TraceTx tx reject)
+  -> TMVar m tx
+  -> m (LocalTxSubmissionClient tx reject m Void)
+txSubmissionClient tracer txv = do
+    tx <- atomically $ readTMVar txv
+    pure $ SendMsgSubmitTx tx $ \mbreject -> do
+      case mbreject of
+        Nothing -> traceWith tracer (TraceTxAccept tx)
+        Just r  -> traceWith tracer (TraceTxReject tx r)
+      txSubmissionClient tracer txv
+
+
+-- | 'ChainSyncClient' which traces received blocks and ignores when it
+-- receives a request to rollbackwar.  A real wallet client should:
+--
+--  * at startup send the list of points of the chain to help synchronise with
+--    the node;
+--  * update its state when the client receives next block or is requested to
+--    rollback, see 'clientStNext' below.
+--
+chainSyncClient
+  :: forall blk m. MonadTimer m
+  => Tracer m blk
+  -> ChainSyncClient blk (Point blk) m Void
+chainSyncClient tracer = ChainSyncClient $ pure $
+    -- Notify the core node about the our latest points at which we are
+    -- synchronised.  This client is not persistent and thus it just
+    -- synchronises from the genesis block.  A real implementation should send
+    -- a list of points up to a point which is k blocks deep.
+    SendMsgFindIntersect
+      [genesisPoint]
```

Comment:
This is a point of confusion that I have begun to clarify in https://github.com/input-output-hk/ouroboros-network/pull/635

If this `SendMsgFindIntersect` were _not_ sent, what would be the read pointer? Wouldn't it still be `genesisPoint`? What if the server does not know `genesisPoint`? Can the protocol continue in that case?

Reply:
If it was not given it would be `genesisPoint` I believe, @mrBliss will know the definite answer.  But the point here is to give the wallet team the starting point.  This is just a hook where they need to provide their state to the local node.

Reply:
Indeed, Genesis is the implicit starting point of the ChainSync protocol (I did not write the protocol). So `SendMsgFindIntersect` is not necessary.